yt_dlp/extractor/twitch.py

```diff
@@ -525,6 +528,82 @@ def _extract_storyboard(self, item_id, storyboard_json_url, duration):
                 } for path in images],
             }
 
+    def _extract_chat(self, vod_id):
```

Comment:
Rename to `_get_subtitles` and call `extract_subtitles` in real_extract. That way, these requests only get called as necessary. See `extract_subtitles` in common.py for how it works.

Reply:
Actually, considering how slow this is, may be better to put this in `__post_extractor` and return the old incomplete subtitles otherwise.

Reply:
should I handle `self.get_param('listsubtitles')` being True in `_get_subtitles`?

Reply:
> and return the old incomplete subtitles otherwise

You mean when `self.get_param('writesubtitles', False)` is False, right?

Reply:
also, it seems to me that `__post_extractor` is called without parameters. how should I know the VOD ID? also, since there is no `self`, how should I call the chat extractor function?

Reply:
Also, why subtitles? Shouldn't it be comments instead?

Reply:
What I mean is, rename this to `_get_subtitles`. Then in `_real_extract`, do this:

```py
        if info.get('timestamp'):
            info['subtitles'] = {'rechat': [{
                'url': update_url_query(f'https://api.twitch.tv/v5/videos/{vod_id}/comments',
                                        {'client_id': self._CLIENT_ID}),
                    'ext': 'json',
            }]}
            info['__post_extractor'] = lambda: {'subtitles': self.extract_subtitles(vid_id)}
```

This way, the extra requests are only made when user requests the subtitles and if --match-filter etc passes. You do not need to access params directly at all - the helpers will handle it.

Reply:
I have restored the old way of downloading the chat, but I think it has not been working for a little more than a year now, when Twitch has shut down their old API.